sql/core/src/main/scala/org/apache/spark/sql/internal/BaseSessionStateBuilder.scala

```diff
@@ -179,6 +179,7 @@ abstract class BaseSessionStateBuilder(
         ResolveEncodersInScalaAgg +:
         new ResolveSessionCatalog(
           catalogManager, conf, catalog.isTempView, catalog.isTempFunction) +:
+        ResolvePartitionSpec +:
```

Comment:
This rule is in catalyst module, we can add it in `Analyzer.scala`, in the `Resolution` batch.